src/NClient/NClient.Standalone/ClientProxy/Generation/Interceptors/ClientInterceptorFactory.cs

```diff
@@ -34,6 +38,13 @@ internal class ClientInterceptorFactory : IClientInterceptorFactory
         private readonly IGuidProvider _guidProvider;
         private readonly IClientRequestExceptionFactory _clientRequestExceptionFactory;
         private readonly IMethodBuilder _methodBuilder;
+        
+        private IPipelineCanceller _pipelineCanceler;
+        public IPipelineCanceller PipelineCanceler
```

Comment:
Each `NClientOptionalBuilder` has one own factory instance. Considering that one builder can create many clients, it turns out that the pipeline cancellation will occur in all created clients and this will create errors. To solve this problem, I suggest passing `IPipelineCanceller` to `ClientInterceptorFactory.Create` method